lib/rucio/core/did.py

```diff
@@ -864,9 +863,11 @@ def list_content_history(scope, name, session=None):
     :param session: The database session in use.
     """
     try:
+        # query = session.query(models.DataIdentifierAssociationHistory).\
+        #    with_hint(models.DataIdentifierAssociationHistory,
+        #              "INDEX(CONTENTS_HISTORY CONTENTS_HIST_PK)", 'oracle').\
+        #    filter_by(scope=scope, name=name)
         query = session.query(models.DataIdentifierAssociationHistory).\
-            with_hint(models.DataIdentifierAssociationHistory,
-                      "INDEX(CONTENTS_HISTORY CONTENTS_HIST_PK)", 'oracle').\
```

Comment:
I would either keep this select together with the relevant index or drop this query without the need for the index. Like this, it goes in a full table scan for each detach operation. Keeping several entries can help to answer the question when files were removed by the production system for example.

Reply:
Yes I fully agree, if there is an API (even CLI command) to access history information, we should have an index. I will add this as a comment for #1818 
The HINT here though should go anyway, as the PK doesnt exist anyway. But as soon as we have an index in the schema (With #1818) I will add it again. (For ATLAS, we actually do have a scope, name index)